jwala-services/src/main/java/com/cerner/jwala/control/webserver/command/WebServerCommandFactory.java

```diff
@@ -71,36 +73,87 @@ public void listCommands() {
+                            webServer)));
+        });
+
+        commands.put(WebServerControlOperation.INSTALL_SERVICE.getExternalValue(), webServer -> {
+            final String installServiceWsScriptName = INSTALL_WS_SERVICE_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(webServer, installServiceWsScriptName);
+            return remoteCommandExecutorService.executeCommand(
+                    new RemoteExecCommand(getConnection(webServer),
+                            getShellCommand(installServiceWsScriptName,
+                                    webServer,
+                                    ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF),
+                                    ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD))));
+        });
+
         commands.put(WebServerControlOperation.VIEW_HTTP_CONFIG_FILE.getExternalValue(), (WebServer webServer)
                 -> remoteCommandExecutorService.executeCommand(
                 new RemoteExecCommand(getConnection(webServer),
-                        new ExecCommand("cat", ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF)+"/httpd.conf"))));
+                        new ExecCommand("cat", ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF) + "/httpd.conf"))));
+
+    }
+
+    private void checkExistsAndCopy(WebServer webserver, String scriptName) {
+        final String destAbsolutePath = getFullPathScript(scriptName, webserver.getName());
+        CommandOutput fileExistsResult = binaryDistributionControlService.checkFileExists(webserver.getHost(), destAbsolutePath);
+        if (!fileExistsResult.getReturnCode().wasSuccessful()) {
+            copyScriptToRemoteDestination(webserver, scriptName, destAbsolutePath);
+        } else {
+            LOGGER.info("{} already exists. Continue with script execution", scriptName);
+        }
+    }
+
+    private void copyScriptToRemoteDestination(WebServer webserver, String scriptName, String destAbsolutePath) {
+        LOGGER.info("{} does not exist at remote location. Performing secure copy.", scriptName);
+
+        final String destDir = new File(getFullPathScript(scriptName, webserver.getName())).getParent();
+        CommandOutput createDirResult = binaryDistributionControlService.createDirectory(webserver.getHost(), destDir);
+        if (!createDirResult.getReturnCode().wasSuccessful()) {
+            LOGGER.error("Failed to create directory {}", destDir);
+            throw new WebServerServiceException("Failed to create directory " + destDir);
+        }
+
+        CommandOutput copyResult = binaryDistributionControlService.secureCopyFile(webserver.getHost(), ApplicationProperties.getRequired("commands.scripts-path") + "/" + scriptName, destAbsolutePath);
+        if (copyResult.getReturnCode().wasSuccessful()) {
```

Comment:
Since the preceding code does
```
if (!success) {
    throw....
}
```
Can we do it here as well ? e.g.
```
if (!copyResult.getReturnCode().wasSuccessful()) {
    LOGGER.error("Failed to secure copy {}", destAbsolutePath);
    throw new WebServerServiceException("Failed to secure copy " + destAbsolutePath);
}
LOGGER.info("Secure copy success to {}", destAbsolutePath);
```